src/views/ThreeWProjectForm/AnnualSplitInput/index.tsx

```diff
@@ -0,0 +1,192 @@
+import React from 'react';
+import {
+    PartialForm,
+    useFormObject,
+    getErrorObject,
+    SetValueArg,
+    Error,
+} from '@togglecorp/toggle-form';
+import {
+    _cs,
+    isDefined,
+    isFalsy,
+} from '@togglecorp/fujs';
+
+import { DeleteBinLineIcon } from '@ifrc-go/icons';
+import NumberInput from '#components/NumberInput';
+import useTranslation from '#hooks/useTranslation';
+import Button from '#components/Button';
+
+import { AnnualSplit } from '../schema';
+import i18n from './i18n.json';
+import styles from './styles.module.css';
+
+type Value = PartialForm<AnnualSplit>;
+
+const defaultValue: Value = {};
+
+interface Props {
+    className?: string;
+    onChange: (value: SetValueArg<Value>, index: number) => void;
+    error: Error<AnnualSplit> | undefined;
+    index: number;
+    value: Value;
+    onRemove: (index: number) => void;
+}
+
+function AnnualSplitInput(props: Props) {
+    const {
+        className,
+        onChange,
+        error: errorFromProps,
+        index,
+        value,
+        onRemove,
+    } = props;
+
+    const strings = useTranslation(i18n);
+
+    const setFieldValue = useFormObject(index, onChange, defaultValue);
+    const error = getErrorObject(errorFromProps);
+
+    // Calculate and set target total
+    React.useEffect(() => {
+        if (
+            isFalsy(value.target_male)
+            && isFalsy(value.target_female)
+            && isFalsy(value.target_other)
+        ) {
+            return;
+        }
+
+        const total = (value.target_male ?? 0)
+            + (value.target_female ?? 0)
+            + (value.target_other ?? 0);
+
+        if (!Number.isNaN(total)) {
+            setFieldValue(total, 'target_total');
+        }
+    }, [value?.target_male, value?.target_female, value?.target_other, setFieldValue]);
```

Comment:
Let's update the handlers to calculate this value instead of relying on useEffect